waspc/data/Generator/templates/react-app/src/auth/forms/Auth.jsx

```diff
@@ -0,0 +1,300 @@
+  position: 'relative',
+  marginTop: '1.5rem'
+})
+
+const OrContinueWithLineContainer = styled('div', {
+  position: 'absolute',
+  inset: '0px',
+  display: 'flex',
+  alignItems: 'center'
+})
+
+const OrContinueWithLine = styled('div', {
+  width: '100%',
+  borderTopWidth: '1px',
+  borderColor: '$gray500'
+})
+
+const OrContinueWithTextContainer = styled('div', {
+  position: 'relative',
+  display: 'flex',
+  justifyContent: 'center',
+  fontSize: '$sm'
+})
+
+const OrContinueWithText = styled('span', {
+  backgroundColor: 'white',
+  paddingLeft: '0.5rem',
+  paddingRight: '0.5rem'
+})
+
+// Email/username form
+
+const UserPassForm = styled('form', {
+  marginTop: '1.5rem'
+})
+
+const FormItemGroup = styled('div', {
+  '& + div': {
+    marginTop: '1.5rem'
+  }
+})
+
+const FormLabel = styled('label', {
+  display: 'block',
+  fontSize: '$sm',
+  fontWeight: '500'
+})
+
+const FormInput = styled('input', {
+  display: 'block',
+  lineHeight: '1.5rem',
+  fontSize: '$sm',
+  borderWidth: '1px',
+  borderColor: '$gray600',
+  backgroundColor: '#f8f4ff',
+  boxShadow: '0 1px 2px 0 rgba(0, 0, 0, 0.05)',
+  '&:focus': {
+    borderWidth: '1px',
+    borderColor: '$gray700',
+    boxShadow: '0 1px 2px 0 rgba(0, 0, 0, 0.05)',
+  },
+
+  borderRadius: '0.375rem',
+  width: '100%',
+
+  paddingTop: '0.375rem',
+  paddingBottom: '0.375rem',
+
+  marginTop: '0.5rem'
+})
+
+const SubmitButton = styled('button', {
+  display: 'flex',
+  justifyContent: 'center',
+
+  width: '100%',
+  borderRadius: '0.375rem',
+  borderWidth: '1px',
+  borderColor: '$brand',
+  backgroundColor: '$brand',
+  color: '$submitButtonText',
+
+  padding: '0.5rem 0.75rem',
+  boxShadow: '0 1px 2px 0 rgba(0, 0, 0, 0.05)',
+
+  fontWeight: '600',
+  fontSize: '$sm',
+  lineHeight: '1.25rem',
+  borderRadius: '0.375rem',
+
+  // TODO(matija): extract this into separate BaseButton component and then inherit it.
+  '&:hover': {
+    backgroundColor: '$brandAccent',
+    borderColor: '$brandAccent',
+  },
+  transitionTimingFunction: 'cubic-bezier(0.4, 0, 0.2, 1)',
+  transitionDuration: '100ms'
+})
+
+const googleSignInUrl = `${config.apiUrl}{= googleSignInPath =}`
+const gitHubSignInUrl = `${config.apiUrl}{= gitHubSignInPath =}`
+
+const Auth = ({ isLogin, appearance, logo, socialLayout }) => {
+  const history = useHistory()
+
+  const [usernameFieldVal, setUsernameFieldVal] = useState('')
+  const [passwordFieldVal, setPasswordFieldVal] = useState('')
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    try {
+      if (!isLogin) {
+        await signup({ username: usernameFieldVal, password: passwordFieldVal })
+      }
+      await login (usernameFieldVal, passwordFieldVal)
+
+      setUsernameFieldVal('')
+      setPasswordFieldVal('')
+
+      // Redirect to configured page, defaults to /.
+      history.push('{= onAuthSucceededRedirectTo =}')
+    } catch (err) {
+      console.log(err)
+      window.alert(errorMessage(err))
```

Comment:
`alerts` are really only useful for debug and they look quite old fashioned. 

Let's please display the message above the form like most other solutions do. 

![image](https://user-images.githubusercontent.com/2223680/229507013-39356fc0-2c7c-4d33-b2de-4847508ad6b2.png)

Reply:
I agree that would be a very nice addition and pizzazz! :)

Reply:
yes, 100%! Maybe I'd only leave it for the end or as a separate issue that somebody else can take, so I can move on to other stuff.

Reply:
agreed with @infomiho he will implement this in a separate PR as he will be adapting the form for email auth